### PR TITLE
bazel: add bazelw wrapper

### PR DESCRIPTION
**Pull request by keith**
This wrapper allows developers to quickly get off the ground without
having to install bazel or bazelisk manually.

Signed-off-by: Keith Smiley <keithbsmiley@gmail.com>

### CODE REVIEW

File: bazelw [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+if [[ ! -x "$bazelisk" ]]; then
+  echo "Installing bazelisk..." >&2
+  mkdir -p "$(dirname "$bazelisk")"
+
+  download_bazelisk() {
+    curl --fail -L --retry 5 --retry-connrefused --silent --progress-bar \
+      --output "$bazelisk" "$bazel_version_url"
+  }
+
+  download_bazelisk || download_bazelisk
```
Comment (jpsim) [inline bazelw:28]: Why retry this on failure when we already have --retry 5 above?

Comment (keith) [reply bazelw:28]: hrm yea, I kinda copied this from our internal script, I'm not sure if one predated the other, or if this attempted to retry a failure that curl didn't retry as intended

Comment (jpsim) [reply bazelw:28]: Can't hurt too much.

**Comment by jpsim** [comment]
Script itself looks good. We'd also need to update scripts and docs.

**Review by jingwei99 — APPROVED** [review]
thank you Keith!

now we just need to scriptify all other 1000 environment setups too (ok there are less than 1000 😂)

**Comment by keith** [comment]
Updated to use for all CI scripts, I don't see public facing docs here that need updating

**Comment by jpsim** [comment]
> I don't see public facing docs here that need updating

I think we can replace the usage of `bazelisk` with `./bazelw` [here](https://github.com/envoyproxy/envoy-mobile/blob/main/docs/root/start/examples/hello_world.rst) and in other files in `docs/`

**Review by buildbreaker — APPROVED** [review]
Thanks @keith !

File: docs/root/development/tools/intellij.rst [hunk 2]
```diff
@@ -10,10 +10,9 @@ Using IntelliJ with Envoy Mobile
 
 To get started using IntelliJ with Envoy Mobile:
 
-1. Locally install `Bazel <https://docs.bazel.build/versions/master/install-os-x.html#install-on-mac-os-x-homebrew>`_
-2. Download a supported `IntelliJ version <https://www.jetbrains.com/idea/download/other.html>`_ supported by the Bazel plugin
-3. Apply local hacks to make IntelliJ work using the branch `hack-for-intellij <https://github.com/lyft/envoy-mobile/tree/hack-for-intellij>`_
-4. Open up the Envoy Mobile project using the Bazel import project wizard
+1. Download a supported `IntelliJ version <https://www.jetbrains.com/idea/download/other.html>`_ supported by the Bazel plugin
+2. Apply local hacks to make IntelliJ work using the branch `hack-for-intellij <https://github.com/lyft/envoy-mobile/tree/hack-for-intellij>`_
```
Comment (Reflejo) [inline docs/root/development/tools/intellij.rst:14]: It was wrong from before but the repo url is https://github.com/envoyproxy/envoy-mobile

Comment (keith) [reply docs/root/development/tools/intellij.rst:14]: https://github.com/envoyproxy/envoy-mobile/pull/1957

**Pull request merged**